cola/utils_test.py

```diff
@@ -36,19 +36,34 @@ def _add_marks(case):
         marks.append(pytest.mark.jax)
     if any('big' in arg for arg in args):
         marks.append(pytest.mark.big)
+    if is_tricky:
+        marks.append(pytest.mark.tricky)
     return pytest.param(*case, marks=marks)
 
 
-def parametrize(*cases, ids=None):
+def parametrize(*cases, tricky=None, ids=None):
     """ Expands test cases with pytest.mark.parametrize but with argnames
-        assumed and ids given by the ids=[str(case) for case in cases] """
+        assumed and ids given by the ids=[str(case) for case in cases] 
+        
+    Certain cases can be marked as tricky, and will be marked with pytest.mark.tricky
+    Tricky can be specified as a list of argument combinations [(arg1,arg2,...), (arg1,arg2,...), ...])])]
+    or as a list of slices [slice(None), slice(None), ["dense","square",...], ...]
+    with lists expanded using the cartesian product (as in cases)."""
     if len(cases) > 1:
         all_cases = [tuple(elem) for elem in itertools.product(*cases)]
     else:
         all_cases = cases[0]
 
+    if tricky is not None:
+        if len(tricky)>1:
+            expanded_tricky = [(c[t] if isinstance(t,slice) else t) for t,c in zip(tricky,cases)]
+            tricky = {tuple(elem) for elem in itertools.product(*expanded_tricky)}
```

Comment:
Not sure what is going on with `tricky` but changing it to `tricky = {tuple(elem) for elem in expanded_tricky}` fixes the issue